sources/core/objects/Floor.cpp

```diff
@@ -36,16 +37,18 @@ void Floor::AttackBy(IdPtr<Item> item)
         if (!open_)
         {
             SetOpen(true);
-            Create<Item>(FloorTile::T_ITEM_S(), GetOwner());
+            IdPtr<FloorTile> ftile = Create<Item>(FloorTile::T_ITEM_S(), GetOwner());
+            ftile->type = floor_type_;
```

Comment:
There is not much sense in `type` in the `FloorTile` because tiles usually stacked (not now in Griefly, but in SS13 they can be stacked). So the unique type for tile will be lost anyway. So only `floor_type_` in `Floor` is needed.